contextvh/src/main/java/contextvh/translators/J2PopupData.java

```diff
@@ -0,0 +1,175 @@
+import nl.tytech.core.client.event.EventManager;
+import nl.tytech.core.net.serializable.MapLink;
+import nl.tytech.core.structure.ItemMap;
+import nl.tytech.data.engine.item.ActionLog;
+import nl.tytech.data.engine.item.Building;
+import nl.tytech.data.engine.item.PopupData;
+import nl.tytech.data.engine.item.Setting;
+import nl.tytech.data.engine.item.SpecialOption;
+import nl.tytech.data.engine.item.SpecialOption.Type;
+import nl.tytech.locale.TCurrency;
+import nl.tytech.locale.unit.UnitSystem;
+import nl.tytech.locale.unit.UnitSystemType;
+import nl.tytech.util.StringUtils;
+
+/**
+ * Translate {@link Popup} into request(type, category, contentlinkID,
+ * [visibleStakeholderIDs], [answers], price, multipolygon).
+ * Multipolygon and price are optional.
+ * @author W.Pasman
+ *
+ */
+public class J2PopupData implements Java2Parameter<PopupData> {
+
+    /**
+     * Translator for additional translations.
+     */
+    private final Translator translator = Translator.getInstance();
+
+    /**
+     * Empty constructor.
+     */
+    public J2PopupData() { }
+
+    /**
+     * Method to translate the PopupData class.
+     */
+    @Override
+    public Parameter[] translate(final PopupData popup) throws TranslationException {
+        String typeOfPopup = null;
+        ParameterList actionLogIds = new ParameterList();
+
+        if (popup.getContentMapLink() == MapLink.SPECIAL_OPTIONS) {
+            SpecialOption specialOption =
+                    EventManager.getItem(MapLink.SPECIAL_OPTIONS, popup.getContentLinkID());
+            Type optionType = specialOption.getType();
+            typeOfPopup = optionType.name();
+        } else if (popup.getContentMapLink() == MapLink.BUILDINGS) {
+            typeOfPopup = "PERMIT";
+            actionLogIds = getActionLogIds(popup);
+
+        } else {
+            typeOfPopup = "POPUP";
+        }
+
+        MultiPolygon mpolygon = popup.getMultiPolygon();
+        Parameter parPolygon = new Identifier("NO_MULTIPOLYGON");
+        Parameter parAreaSize = new Identifier("NO_SIZE");
+        if (mpolygon != null) {
+            parPolygon = translator.translate2Parameter(mpolygon)[0];
+            parAreaSize = new Numeral(mpolygon.getArea());
+        }
+        Double price = getPriceFromPopup(popup);
+        Parameter parPrice = new Identifier("NO_PRICE");
+        if (price != null) {
+            parPrice = new Numeral(price);
+        }
+
+        return new Parameter[] {new Function("request",
+                new Identifier(popup.getType().name()),
+                new Identifier(typeOfPopup),
+                new Numeral(popup.getID()),
+                new Numeral(popup.getContentLinkID()),
+                getVisibleForStakeholderIDs(popup.getVisibleForStakeholderIDs()),
+                actionLogIds,
+                parPrice,
+                parPolygon,
+                parAreaSize,
+                translator.translate2Parameter(popup.getAnswers())[0])};
```

Comment:
@daanwerf please refer to the J2Popupdata at this position for the parameters of the percept if you need to look up what it's made of.